File: output/csbc_semish_2026/deepseek_deepseek_v3_2/zero/cpp/Chocolate em barra.cpp

#include <iostream>
using namespace std;

int main() {
    int N, x1, y1, x2, y2;
    cin >> N >> x1 >> y1 >> x2 >> y2;
    
    // Verifica se é possível separar as figurinhas com um corte vertical
    bool vertical_possible = (x1 <= N/2 && x2 > N/2) || (x2 <= N/2 && x1 > N/2);
    
    // Verifica se é possível separar as figurinhas com um corte horizontal
    bool horizontal_possible = (y1 <= N/2 && y2 > N/2) || (y2 <= N/2 && y1 > N/2);
    
    if (vertical_possible || horizontal_possible) {
        cout << "S" << endl;
    } else {
        cout << "N" << endl;
    }
    
    return 0;
}

File: output/csbc_semish_2026/deepseek_deepseek_v3_2/zero/python/Chocolate em barra.py

import sys

def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    N = int(data[0])
    x1, y1 = int(data[1]), int(data[2])
    x2, y2 = int(data[3]), int(data[4])

    # Verifica se é possível separar as figurinhas com um corte vertical
    if (x1 <= N // 2 and x2 > N // 2) or (x2 <= N // 2 and x1 > N // 2):
        print("S")
        return

    # Verifica se é possível separar as figurinhas com um corte horizontal
    if (y1 <= N // 2 and y2 > N // 2) or (y2 <= N // 2 and y1 > N // 2):
        print("S")
        return

    print("N")

if __name__ == "__main__":
    main()
